### PR TITLE
Adds an agent that plays by the book

### PR DESCRIPTION
**Pull request by dtumer**
This change adds an agent that plays by the book

### CODE REVIEW

File: src/agents/players/book_agent.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay]
+    ]
+
+    def __init__(self):
+        super().__init__()
+
+        self.dealer_card = 0
+
+    def update_model(self, game):
+        self.dealer_card = game.dealer.hands[0].cards[1]
```
Comment (rsmith49) [inline src/agents/players/book_agent.py:187]: I think this should be `cards[0]`

File: src/agents/players/book_agent.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    by the book action matrix for hands that aren't soft but we can split
+        1  2  3  4  5  6  7  8  9  10  
+    4   H  H  H  H  H  H  H  H  H  H
+    6   H  H  H  H  H  H  H  H  H  H
+    8   H  H  H  H  H  H  H  H  H  H
+    10  H  D  D  D  D  H  H  H  H  H
+    12  H  H  H  H  H  H  H  H  H  H
+    14  P  P  P  P  P  P  P  P  P  P
+    16  P  P  P  P  P  P  P  P  P  P
+    18  P  P  P  P  P  P  P  P  P  P
```
Comment (rsmith49) [inline src/agents/players/book_agent.py:164]: Some of the optimal strategies I've seen vary, but none on the first few results of Google images say to split 9's or 7's all the time. What guide are you using for optimal strategy?

File: src/agents/players/book_agent.py (hunk 1 above)
Comment (dtumer) [reply src/agents/players/book_agent.py:187]: is it the first or second card that is dealt that the player sees?

File: src/agents/players/book_agent.py (hunk 2 above)
Comment (dtumer) [reply src/agents/players/book_agent.py:164]: I didn't haha. I'll update this. Pretty much guessed at the strategy given what I remember. I'll fix it to include a real one I find online

File: src/agents/players/book_agent.py (hunk 1 above)
Comment (rsmith49) [reply src/agents/players/book_agent.py]: First

**Pull request merged**